src/com/sittinglittleduck/DirBuster/Worker.java

```diff
@@ -573,20 +321,176 @@ else if(work.getBaseCaseObj().isUseRegexInstead())
+             * Host header has to be set in a different way!
+             */
+            if(httpHeader.getHeader().startsWith("Host"))
+            {
+            	httpMethod.getParams().setVirtualHost(httpHeader.getValue());
+            }
+            else
+            {
+            	httpMethod.setRequestHeader(httpHeader.getHeader(), httpHeader.getValue());
             }
+
         }
+        httpMethod.setFollowRedirects(Config.followRedirects);
 
+        /*
+         * this code is used to limit the number of request/sec
+         */
+        if(manager.isLimitRequests())
+        {
+            while(manager.getTotalDone() / ((System.currentTimeMillis() - manager.getTimestarted()) / 1000.0) > manager.getLimitRequestsTo())
+            {
+                Thread.sleep(100);
+            }
+        }
+        /*
+         * Send the head request
+         */
+        return httpclient.executeMethod(httpMethod);
     }
 
+	private boolean verifyIfCodeIsValid(int code) {
+		return code != HttpStatus.SC_NOT_FOUND && code != 0 && code != HttpStatus.SC_BAD_GATEWAY;
+	}
+
+	private void verifyResponseForOtherRequests(int code, String responce, String rawResponce) {
+		if(work.isDir())
+		{
+		    if(Config.debug)
+		    {
+		        System.out.println("DEBUG Worker[" + threadId + "]: Found Dir (base case)" + url.toString());
+		    }
+		    //we found a dir
+		    manager.foundDir(url, code, responce, work.getBaseCaseObj().getBaseCase(), rawResponce, work.getBaseCaseObj());
+		}
+		else
+		{
+		    //found a file
+		    if(Config.debug)
+		    {
+		        System.out.println("DEBUG Worker[" + threadId + "]: Found File (base case)" + url.toString());
+		    }
+		    manager.foundFile(url, code, responce, work.getBaseCaseObj().getBaseCase(), rawResponce, work.getBaseCaseObj());
+		}
+               //manager.foundError(url, "Base Case Mode Error - Responce code came back as " + code + " it should have been 200");
+               //manager.workDone();
+	}
+
+	private void verifyResponseForValidRequests(int code, String responce, String rawResponce) {
+		if(Config.debug)
+		{
+		    System.out.println("DEBUG Worker[" + threadId + "]: Base Case Check " + url.toString());
+		}
+
+
+		//TODO move this option to the Adv options
+		//if the response does not match the base case
+		Pattern regexFindFile = Pattern.compile(".*file not found.*", Pattern.CASE_INSENSITIVE);
+
+		Matcher m = regexFindFile.matcher(responce);
+
+		//need to clean the base case of the item we are looking for
+		String basecase = FilterResponce.removeItemCheckedFor(work.getBaseCaseObj().getBaseCase(), work.getItemToCheck());
+
+		if(m.find())
+		{
+			System.out.println("DEBUG Worker[" + threadId + "]: 404 for: " + url.toString());
+		}
+		else if(!responce.equalsIgnoreCase(basecase))
+		{
+		    sendResponseToManager(code, responce, rawResponce, basecase);
+		}
+	}
+
+	private void sendResponseToManager(int code, String responce, String rawResponce, String basecase) {
```

Comment:
I would call this method ```notifyItemFound``` or something like that, that's want the method is doing, notify the manager that a directory/file was found.
The same comment as above regarding the typos.